### PR TITLE
fix: updating CNI to v1.5.9 for dropgz

### PR DESCRIPTION
**Pull request by rjdenney**
**Reason for Change**:

Updating CNI version in dropgz to CNI v1.5.9 that has a fix for preventing the CNS to panic when no NCs are found on an NNC.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by camrynl — DISMISSED** [review]

File: dropgz/build/linux.Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ COPY ./azure-ipam .
 RUN curl -LO --cacert /etc/ssl/certs/ca-certificates.crt https://github.com/Azure/azure-container-networking/releases/download/azure-ipam%2F$AZIPAM_VERSION/azure-ipam-$OS-$ARCH-$AZIPAM_VERSION.tgz && tar -xvf azure-ipam-$OS-$ARCH-$AZIPAM_VERSION.tgz
 
 FROM tar AS azure-vnet
-ARG AZCNI_VERSION=v1.5.4
+ARG AZCNI_VERSION=v1.5.7
```
Comment (paulyufan2) [inline dropgz/build/linux.Dockerfile:15]: should not we use v1.5.8 now?

Comment (rjdenney) [reply dropgz/build/linux.Dockerfile:15]: Yeah we are going to end up using v1.5.8 once it is ready. Once we have it I'll update this to reflect v1.5.8

File: dropgz/build/linux.Dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@ COPY ./azure-ipam .
 RUN curl -LO --cacert /etc/ssl/certs/ca-certificates.crt https://github.com/Azure/azure-container-networking/releases/download/azure-ipam%2F$AZIPAM_VERSION/azure-ipam-$OS-$ARCH-$AZIPAM_VERSION.tgz && tar -xvf azure-ipam-$OS-$ARCH-$AZIPAM_VERSION.tgz
 
 FROM tar AS azure-vnet
-ARG AZCNI_VERSION=v1.5.4
+ARG AZCNI_VERSION=v1.5.8
```
Comment (vipul-21) [inline dropgz/build/linux.Dockerfile:15]: We recently added the dropgz for windows too. So would be better to update the version for that as well: https://github.com/Azure/azure-container-networking/blob/master/dropgz/build/windows.Dockerfile#L9

**Review by ramiro-gamarra — DISMISSED** [review]

**Review by paulyufan2 — DISMISSED** [review]
lgtm

**Review by ramiro-gamarra — APPROVED** [review]

**Review by vipul-21 — APPROVED** [review]

**Pull request merged**